server/src/main/java/org/elasticsearch/action/admin/indices/rollover/Condition.java

```diff
@@ -67,6 +67,10 @@ public String name() {
         return name;
     }
 
+    public boolean isRequired() {
```

Comment:
I also think it's kind of a difficult/confusing name, perhaps "requiresAdditionalConditions" or "requiresMaxCondition" instead?

Reply:
How about inverting this -- rather than a boolean method that says "here's meaning" how about returning an Enum for "here's what this is"? That is, something like `Condition.Type.MIN` and `Condition.Type.MAX`.

We care about this in two places -- `validate()` (has to have at least one MAX), and also when evaluating `areConditionsMet` (all MIN conditions and at least one MAX condition must be 'met'). It seems like both those cases would read clearer if they just were what they said, rather than abstracting away behind some boolean method.

It'd be more direct, but it would also be tighter binding. That said, though, I don't see any reason we couldn't revisit that in the future if some third kind of condition or new rules, etc, were added.

Reply:
That would be totally okay with me too, my main concern is that from the name alone, there is no way to tell what this actually signifies, or how it is used. Returning a condition type would work for differentiating both of those, so that works for me.

Reply:
👍, 8a60331e4f0fc7a953d3b5eaf326afa083d5afea